core/src/main/java/org/apache/brooklyn/entity/group/DynamicClusterImpl.java

```diff
@@ -181,6 +189,34 @@ private void initialiseMemberId() {
         }
     }
 
+    private void connectAllMembersUp() {
+        allMembersUp = FunctionFeed.builder()
+                .entity(this)
+                .period(Duration.FIVE_SECONDS)
+                .poll(new FunctionPollConfig<Boolean, Boolean>(ALL_MEMBERS_UP)
+                        .onException(Functions.constant(Boolean.FALSE))
+                        .callable(new AllMembersUpCallable()))
+                .build();
+    }
+
+    private class AllMembersUpCallable implements Callable<Boolean> {
+
+        @Override
+        public Boolean call() throws Exception {
+            if (DynamicClusterImpl.this.getMembers().isEmpty())
+                return false;
+
+            if (Lifecycle.RUNNING != DynamicClusterImpl.this.sensors().get(SERVICE_STATE_ACTUAL))
+                return false;
+
+            for (Entity member : DynamicClusterImpl.this.getMembers())
+                if (Boolean.FALSE.equals(member.sensors().get(SERVICE_UP)))
```

Comment:
The callable is named `AllMembersUp` but this is checking that all members are not not up. It will return true if a member's `SERVICE_UP` is null. I think you should swap the if condition for `if (!Boolean.TRUE.equals(...))`.